### ISSUE TITLE
PHPStan mixin recursion? 

### ISSUE THREAD

**Issue by sanfair** [barryvdh/laravel-ide-helper#1155]
### Versions:
- ide-helper Version: 2.9.0
- Laravel Version: 8.20.1
- Larastan Version: 0.6.12
- PHPStan Version: 0.12.68
- PHP Version: 8.0.2

### Question:

When generate autocompletion for models  using command `php artisan ide-helper:models -M` it will generate _ide_helper_models.php file with next code

```php
namespace App\Models{
/**
 * App\Models\City
 *
 * @mixin IdeHelperCity
 * @property int $id
 * @property int $country_id
 * @property string $name
 * @property \Illuminate\Support\Carbon|null $created_at
 * @property \Illuminate\Support\Carbon|null $updated_at
 * @property \Illuminate\Support\Carbon|null $deleted_at
 * @property-read \App\Models\Country $country
 * @property-read \Illuminate\Database\Eloquent\Collection|\App\Models\Office[] $offices
 * @property-read int|null $offices_count
 * @method static Builder|City filter(array $filters)
 * @method static Builder|City newModelQuery()
 * @method static Builder|City newQuery()
 * @method static \Illuminate\Database\Query\Builder|City onlyTrashed()
 * @method static Builder|City query()
 * @method static Builder|City whereCountryId($value)
 * @method static Builder|City whereCreatedAt($value)
 * @method static Builder|City whereDeletedAt($value)
 * @method static Builder|City whereId($value)
 * @method static Builder|City whereName($value)
 * @method static Builder|City whereUpdatedAt($value)
 * @method static \Illuminate\Database\Query\Builder|City withTrashed()
 * @method static \Illuminate\Database\Query\Builder|City withoutTrashed()
 */
	class IdeHelperCity extends \Eloquent {}
}
```

Can someone explain why do we need `@mixin IdeHelperCity` here?

I think this cause phpstan run in infinite recursion trying analyse this class.
Without this line phpstan works fine

**Comment by mfn** [comment]
I don't recall, but I'm also not using `_ide_helper_models.php` (I just write the meta data to the models directly which does not suffer this problem it seems? and works fine with phpstan/larastan).

Maybe you can track this in the git history how/why.

**Comment by sanfair** [comment]
> Maybe you can track this in the git history how/why.

Yeah i did check it, it was added when implemented this feature (#764)

This is nice feature, because we don't need maintain phpDoc

**Comment by edcoreweb** [comment]
@mfn @sanfair I am not sure if this is a bug or a feature :D

So how things actually work: the model docblock is read, stuff is added and then is written to the `_ide_helper_models.php` file.
The first time the command is executed, there's no `@mixin IdeHelperCity`, the second time it's there and get's picked up for the journey.

There's a option `--reset` which basically discards what's on the class docblock.
This would fix your issue.

If we would want to fix the undelaying issue, the ```@mixin IdeHelperCity``` tag needs to be removed before the phpdoc is serialized
https://github.com/barryvdh/laravel-ide-helper/blob/c1ddd3053205c56a669da2bf926fc873856e7401/src/Console/ModelsCommand.php#L876

Maybe something like this:

```php
if ($this->write_mixin) {
    $phpdocMixin = new DocBlock($reflection, new Context($namespace));
    ...
    ...
    // remove all mixin tags prefixed with IdeHelper from the main phpdoc
    foreach ($phpdoc->getTagsByName('mixin') as $tag) {
        if (Str::startsWith($tag->getContent(), 'IdeHelper')) {
            $phpdocMixin->deleteTag($tag);
        }
    }
}

$docComment = $serializer->getDocComment($phpdoc);

```
